code-challenges.test.js

```diff
@@ -10,45 +12,152 @@
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-var testArray1 = [3, 7, 10, 5, 4, 3, 3]
-var testArray2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
+// 3a) Pseudo code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
 
+describe("uniqueArray", () => {
+  it("A function which takes in two arrays as arguments and returns one array with no duplicated values", () => {
+    expect(uniqueArray([3, 7, 10, 5, 4, 3, 3], [7, 8, 2, 3, 1, 5, 4])).toEqual([
+      3, 7, 10, 5, 4, 8, 2, 1,
+    ]);
+  });
+});
 
 // b) Create the function that makes the test pass.
+
+// 3b) Pseudo code:
+// Declare a function and name it as uniqueArray (const declaration)
+// Name two parameters as arr1 and arr2
+// Declare a variable, name it newArray, and initialize it as a concatenation of both array arguments
+// Use set() to modify newArray and only return unique values (have newArray equal to itself)
+// return newArray
+// Run the test to verify it passes green
+
+const uniqueArray = (arr1, arr2) => {
+  let newArray = arr1.concat(arr2);
+  newArray = new Set(newArray);
+  newArray = [...newArray];
+  return newArray;
+};
+
+// Notes: I ran the intial test and got the right red failure. While implementing the code, I had to try a few different methods for getting set to work. This is because Set() creates an object, not an array (I had some syntax issues to troubleshoot). But this definitely seems quicker than using a forloop, which is what I used last time I saw this problem. I got the code to pass green. I would make this a one liner, but I'm not really sure how. I think Ill leave this as is. Overall, it was cool learning how to use Set(). They definitely seem to have some nice applications.
```

Comment:
Nice work. Yeah `Set` is a great tool. Definitely cleaner than looping. I think you could refactor a bit but I wouldn't do too much. Readable code is always the best option.